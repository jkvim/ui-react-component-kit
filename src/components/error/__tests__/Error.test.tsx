import * as React from 'react';
import { mount } from 'enzyme';
import toJSON from 'enzyme-to-json';
import { Error } from '../';

describe('Error', () => {

  describe('props', () => {

    const wrapper = mount(
      <Error id="id" message="error" />);

    it('should match snapshot', () => {
      expect(toJSON(wrapper)).toMatchSnapshot();
    });

    it('should pass `id` in props', () => {
      expect(wrapper.props().id).toBe('id');
    });

    it('should pass `message` in props', () => {
      expect(wrapper.props().message).toBe('error');
    });
  });
});