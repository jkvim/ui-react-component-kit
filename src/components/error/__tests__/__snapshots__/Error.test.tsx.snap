// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Error props should match snapshot 1`] = `
<Error
  id="id"
>
  <styled.div
    id="id-error"
    theme={
      Object {
        "border": Object {
          "radius": Object {
            "edge": "2px",
            "round": "100px",
          },
          "size": "2px",
        },
        "breakpoint": Object {
          "lg": "992px",
          "md": "768px",
          "sm": "576px",
          "xl": "1200px",
          "xs": "0px",
        },
        "color": Object {
          "background": "#ffffff",
          "border": "#2fa4e7",
          "checked": "#2fa4e7",
          "danger": "#c71c22",
          "default": "#2fa4e7",
          "disabled": "#acacac",
          "error": "#c71c22",
          "focus": "#54b4eb",
          "font": "#333333",
          "hover": "#178acc",
          "info": "#04519b",
          "primary": "#2fa4e7",
          "secondary": "#e9ecef",
          "success": "#73a839",
          "tertiary": "#343a40",
          "warning": "#dd5600",
        },
        "font": Object {
          "size": Object {
            "default": "1.8rem",
            "h1": "2.8rem",
            "h2": "2.1rem",
            "h3": "1.6rem",
            "h4": "1.3rem",
            "h5": "1.2rem",
            "h6": "1.1rem",
          },
          "weight": Object {
            "default": 100,
          },
        },
      }
    }
  >
    <div
      className="sc-bdVaJa lhSjwC"
      id="id-error"
    >
      error
    </div>
  </styled.div>
</Error>
`;
